test/arithmetic/floating_point/floating_point_conversion_test.dart

```diff
@@ -0,0 +1,278 @@
+      destExponentWidth: bf19ExponentWidth,
+      destMantissaWidth: bf19MantissaWidth,
+    );
+
+    final packedFPbf16 = FloatingPointConverter.convertNormalNumber(
+      source: fp32,
+      destExponentWidth: bf16ExponentWidth,
+      destMantissaWidth: bf16MantissaWidth,
+    );
+
+    final packedFPfp16 = FloatingPointConverter.convertNormalNumber(
+      source: fp32,
+      destExponentWidth: fp16ExponentWidth,
+      destMantissaWidth: fp16MantissaWidth,
+    );
+
+    final packedFPbf8 = FloatingPointConverter.convertNormalNumber(
+      source: fp32,
+      destExponentWidth: bf8ExponentWidth,
+      destMantissaWidth: bf8MantissaWidth,
+    );
+
+    final packedFPhf8 = FloatingPointConverter.convertNormalNumber(
+      source: fp32,
+      destExponentWidth: hf8ExponentWidth,
+      destMantissaWidth: hf8MantissaWidth,
+    );
+
+    expect(packedFPbf19.isNormal().value.toBool(), true);
+    expect(packedFPbf16.isNormal().value.toBool(), true);
+    expect(packedFPfp16.isNormal().value.toBool(), true);
+    expect(packedFPbf8.isNormal().value.toBool(), true);
+    expect(packedFPhf8.isNormal().value.toBool(), true);
+  });
+
+  test('FP: converter builds', () {
+    final fp32 = FloatingPoint32()
+      ..put(FloatingPoint32Value.fromDouble(1.512312312312312).value);
+
+    final converter = FloatingPointConverter(fp32,
+        destExponentWidth: bf19ExponentWidth,
+        destMantissaWidth: bf19MantissaWidth,
+        name: 'fp32_to_bf19');
+
+    expect(converter.result.floatingPointValue.toString().contains('x'), false);
+    expect(converter.result.floatingPointValue.toString().contains('z'), false);
+  });
+
+  test('FP: converter processes TestFloat vectors', () async {
+    // Run testfloat_gen and capture its output
+    // Get the current test file path
+    final currentFilePath = Platform.script.toFilePath();
+    print('Current test file: $currentFilePath');
+
+    // Get the directory containing the test file
+    final testDir = File(currentFilePath).parent.path;
+    print('Test directory: $testDir');
+
+    // Run testfloat_gen from a specific directory
+    final result = await Process.run(
+      'testfloat_gen',
+      ['f32_to_f16'],
+      workingDirectory: testDir,
+    );
+    if (result.exitCode != 0) {
+      throw Exception('TestFloat failed: ${result.stderr}');
+    }
+
+    final testVectors = (result.stdout as String).split('\n');
+
+    for (final line in testVectors) {
+      if (line.trim().isEmpty) continue;
+
+      final parts = line.trim().split(RegExp(r'\s+'));
+      final inputHex = parts[0];
+      final expectedHex = parts[1];
+      final flags = parts[2];
+
+      final inputBits = int.parse(inputHex, radix: 16);
+
+      final fp32 = FloatingPoint32()..put(inputBits);
+
+      final converter = FloatingPointConverter(fp32,
+          destExponentWidth: fp16ExponentWidth,
+          destMantissaWidth: fp16MantissaWidth,
+          name: 'fp32_to_bf19');
+
+      print('Test case:');
```

Comment:
you can use the `reason` field of `expect` to give more detailed explanations, and `'''`s to make multi-line strings